app/src/main/java/free/rm/skytube/businessobjects/YouTube/POJOs/YouTubeChannel.java

```diff
@@ -247,7 +249,8 @@ public boolean blockChannel(boolean displayToastMessage) {
 
 		// if user is subscribed to the channel, then unsubscribe first...
 		if (SubscriptionsDb.getSubscriptionsDb().isUserSubscribedToChannel(getId())) {
```

Comment:
I think, this 'isUserSubscribedToChannel' is also a db call, which could block.

Reply:
In that case, it could be converted to a `Single`.